internal/api/metadata/describe_test.go

```diff
@@ -0,0 +1,25 @@
+package metadata
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+func TestMetadataDescribe(test *testing.T) {
+	response := core.SendTestAPIRequest(test, `metadata/describe`, nil)
+	if !response.Success {
+		test.Fatalf("Response is not a success when it should be: '%v'.", response)
+	}
+
+	var responseContent DescribeResponse
+	util.MustJSONFromString(util.MustToJSON(response.Content), &responseContent)
+
+	expected := DescribeResponse{*core.Describe(core.GetServerRoutes())}
+	if !reflect.DeepEqual(expected, responseContent) {
+		test.Fatalf("Unexpected API description. Expected: '%s', actual: '%s'.",
+			util.MustToJSONIndent(expected), util.MustToJSONIndent(responseContent))
+	}
```

Comment:
The real test for this will probably need to compare against an actual, hand-checked JSON object.
This is a design thing to think about: how will we maintain this object.
It could be nice to just keep a JSON file in the repo that is the output of cmd/describe.
This will allow things (like this test and the Python side) to just grab this file whenever it needs to.
But, then the file needs to be maintained or this test will fail.
Maybe we have a script to automatically updates the file?
(Note that the script would be something super simple like: `go run cmd/describe-api/main.go > api.json`.)